test/auto_parallel/spmd_rules/test_reshape_rule.py

```diff
@@ -248,6 +248,171 @@ def test_reshape_infer_forward(self):
         with self.assertRaises(BaseException):
             self.rule.infer_forward([self.x_dist_tensor_spec], self.attrs)
 
+    def test_reshape_infer_backward(self):
```

Comment:
unitest lack some cases where the axis of output  is shard and need to be recover to unsharded before reshard：

case1: last axis of output
shape: [6, 12, 48, 24] --> [1, 72, 48, 4, 6] (input --> output)
dims_mapping: [-1, 0, -1, -1, 1] --> [0, -1, -1, -1], [-1, 0, 1, -1, -1] (output --> input, output)

case2: 2nd axis of output
shape: [6, 12, 48, 24] --> [3, 24, 6, 8, 24] (input --> output)
dims_mapping: [-1, 1, -1, -1, 0] --> [-1, -1, -1, 0], [-1, -1, -1, -1, 0] (output --> input, output)